### ISSUE TITLE
[Bug]: Multiple labelfilters do not combine correctly

### ISSUE THREAD

**Issue by jasonckchan** [jasp-stats/jasp-issues#2376]
### JASP Version

0.18.1

### JASP Module

Other

### What analysis are you seeing the problem on?

All tests I've tried

### What OS are you seeing the problem on?

macOS Silicon

### Bug Description

When I tried to apply filters for multiple variables, instead of the filters stacking, the filters will somehow interfere with each other and not stack.  The only way for multiple filters to apply correct (stacking, or intersecting) is to use the "easy filter" application, in which I drag variables names into the filter space, add the equal (or unequal) sign, and then typing in what I want.  If I double click on the variable names to create filters, I cannot stack the filters.  In fact, the filtered results would be very weird.
[NSF-SL Study 1a Data FINAL.csv](https://github.com/jasp-stats/jasp-issues/files/13221572/NSF-SL.Study.1a.Data.FINAL.csv)
![Screenshot 2023-10-31 at 4 43 09 PM](https://github.com/jasp-stats/jasp-issues/assets/78377371/86f7d5e5-c3db-4a43-b10c-1e5d87eb5cd5)

### Expected Behaviour

Filters should stack.

### Steps to Reproduce

1. Double click the "Topic" variable and deselect "ECO."
2. Double click the "Distractor" variable and deselect "MM" and "VD." 
3. You'll see that JASP would not filter the results correctly.  Neither the Topic variable nor the Distractor variable will be filtered correctly.
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by JorisGoosen** [comment]
Ooh no!

We'll get on it

**Comment by tomtomme** [comment]
I can confirm this for the flatpak version

**Comment by JorisGoosen** [comment]
Ok Ive solved the bug, so once the fix is merged we can start getting ready for a good bugfixing release.

**Comment by bfmalle** [comment]
I assume the bugfixing release hasn't happened?  Because the filter problem persists for me.  There are related problems, where filters stay on even though I have unclicked them (namely, when there is a filter applied to some other column already), and the filtering doesn't seem to work at all when I manually relabeled the values of a variable (e.g., from 0, 1 to Democrat, Republican).

**Comment by boutinb** [comment]
Yes this is solved, but we are still in testing phase. The version should be released soon. You can try this beta version if you like: https://static.jasp-stats.org/Nightlies/macOS-arm64/JASP-nightly-development-8252e79adccbefddc70582a14879a792356bffc7.dmg

**Comment by rnorlund** [comment]
The ability to use multiple labelfilters is still not working properly as of March 10, 2024. 
And the link above does not work for the beta version. 
Could you repost please as this ability is pretty critical to a student's project! 
Thank you!
Roger

**Comment by tomtomme** [comment]
It should work for analyses, but it is broken for computed columns. Correct?

**Comment by tomtomme** [comment]
See https://github.com/jasp-stats/jasp-issues/issues/2553

**Comment by Neelik** [comment]
I can confirm that as of March 11, 2024, for a nominal column, attempting to filter out two labels fails. I am not sure if you would consider this as "multiple label filters", so if not I can make a new issue if necessary.

**Comment by tomtomme** [comment]
@Neelik 
I do not have that problem here on linux. So, yes, please make a new issue including the .jasp file that shows this. Also including additional infos like your OS, version etc. 
The more details the better. Than we may be able to fix this.
Thanks!

**Comment by JorisGoosen** [comment]
@Neelik the date is not particularly helpful...

As @tomtomme mentions giving us some more info like OS and jasp version would be the minimal information we need to be able to help you at all.
